### PR TITLE
Assorted work on the build infrastructure

### PR DESCRIPTION
**Pull request by gsmet**
Fixes:
https://hibernate.atlassian.net/browse/OGM-1056
https://hibernate.atlassian.net/browse/OGM-1057
https://hibernate.atlassian.net/browse/OGM-1058
https://hibernate.atlassian.net/browse/OGM-1059

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -26,12 +24,16 @@ install:
   # The Maven install provided by Travis is outdated, use Maven wrapper to get the latest version
   - mvn -N io.takari:maven:wrapper
   - ./mvnw -v
+  # Use the Google mirror for central - we also declare explicitly central as the JBoss repository takes precedences over it
+  - sed -i settings-example.xml -e "s@</mirrors>@<mirror><id>google-maven-central</id><name>Google Maven Central</name><url>https://maven-central.storage.googleapis.com</url><mirrorOf>central</mirrorOf></mirror></mirrors>@"
+  - sed -i settings-example.xml -e "s@<repositories>@<repositories><repository><id>central</id><name>Central</name><url>https://maven-central.storage.googleapis.com/</url></repository>@"
+  - sed -i settings-example.xml -e "s@<pluginRepositories>@<pluginRepositories><pluginRepository><id>central</id><name>Central</name><url>https://maven-central.storage.googleapis.com/</url></pluginRepository>@"
   # first run to download all the Maven dependencies without logging
-  - travis_wait ./mvnw -B -q -s settings-example.xml -Ptest -DskipTests=true install
+  - travis_wait ./mvnw -B -q -s settings-example.xml -Ptest -DskipTests=true -Dmaven.javadoc.skip=true -DskipDistro=true install
```
Comment (DavideD) [inline .travis.yml]: I think you can skip the whole documentation project using the `-DskipDocs` parameter. Not sure if in this case makes a difference

File: pom.xml [hunk 2]
```diff
@@ -618,22 +619,14 @@
...
-                    <version>0.1.13</version>
+                    <version>0.3.4</version>
                     <configuration>
                         <port>${embeddedMongoDbPort}</port>
                         <logging>file</logging>
                         <logFile>${embeddedMongoDbLogDir}/embedded-mongodb.log</logFile>
                         <bindIp>${embeddedMongoDbBindIp}</bindIp>
                         <version>${embeddedMongoDbVersion}</version>
                     </configuration>
-                    <dependencies>
```
Comment (gunnarmorling) [inline pom.xml]: Ah, nice.

**Comment by gunnarmorling** [comment]
Rebased and applied. I like the plug-in updates at this point. Mostly they mean a speed-up or bug fixes, and we always can re-adjust if needed. Thanks, @gsmet!

File: .travis.yml (hunk 1 above)
Comment (gsmet) [inline .travis.yml]: In fact, I want to skip the generation but I still want it to download as much plugin dependencies as possible. That's why it's done like that!

Comment (DavideD) [inline .travis.yml]: I thought so

**Pull request closed without merging**